staging/src/k8s.io/cli-runtime/pkg/resource/dry_run_verifier.go

```diff
@@ -29,15 +29,15 @@ import (
 
 // VerifyDryRun returns nil if a resource group-version-kind supports
 // server-side dry-run. Otherwise, an error is returned.
-func VerifyDryRun(gvk schema.GroupVersionKind, dynamicClient dynamic.Interface, discoveryClient discovery.DiscoveryInterface) error {
-	verifier := NewDryRunVerifier(dynamicClient, discoveryClient)
+func VerifyDryRun(gvk schema.GroupVersionKind, dynamicClient dynamic.Interface, openAPIGetter discovery.OpenAPISchemaInterface) error {
```

Comment:
Couldn't we just get the schema here? I don't know of the top of my head, but don't we always get the schema before creating the VerifyDryRun, why even using the lazy mechanism?

Reply:
Ah yes we can do that. No reason why it's lazy

Reply:
I found why it's lazy: we don't want to always get the OpenAPI document for every command where we implement dry run. Tests break when we fix this, and we'd have to inject a discovery.OpenAPISchemaInterface properly for tests where we don't need the document